src/index.ts

```diff
@@ -1,92 +1,81 @@
 import cluster from "node:cluster";
-import http from "node:http";
+import { Bot } from "grammy";
+import dotenv from "dotenv";
 
-const PORT = 3000;
+dotenv.config({ path: ".env.local" });
+if (!process.env.BOT_TOKEN_ONE) {
+  throw new Error("BOT_TOKEN_ONE is not defined");
+}
+if (!process.env.BOT_TOKEN_TWO) {
+  throw new Error("BOT_TOKEN_TWO is not defined");
+}
 
-if (cluster.isPrimary) {
-  console.log(`Primary ${process.pid} is running`);
+interface WorkerMessage {
+  type: "INITIALIZE_BOT" | "SHUTDOWN";
+  token?: string;
+}
 
-  for (let i = 0; i < 3; i++) {
-    cluster.fork();
-  }
+// Your bot configurations
+const bots: string[] = [process.env.BOT_TOKEN_ONE, process.env.BOT_TOKEN_TWO];
 
-  cluster.on("exit", (worker, code, signal) => {
-    console.log(
-      `Worker ${worker.process.pid} died with code: ${code} and signal: ${signal}`,
-    );
+if (cluster.isPrimary) {
+  const workerBotTokens = new Map();
 
-    console.log("Starting a new worker");
-    cluster.fork();
+  // Map a worker to a bot
+  bots.forEach((bot, _) => {
+    const worker = cluster.fork();
+    workerBotTokens.set(worker.id, bot);
   });
 
-  const server = http.createServer((req, res) => {
-    res.writeHead(200);
-    res.end(`Hello from Primary ${process.pid}\n`);
-
-    if (cluster.workers) {
-      try {
-        const workerIds = Object.keys(cluster.workers);
-        const workerId =
-          workerIds[Math.floor(Math.random() * workerIds.length)];
-        const worker = cluster.workers[workerId];
-
-        if (!worker) return;
-
-        try {
-          worker.send("request");
-          console.log(
-            `Delegated background processing to Worker ${worker.process.pid}`,
-          );
-        } catch (error: unknown) {
-          console.error(
-            `Failed to send message to worker ${worker.process.pid}:`,
-            error instanceof Error ? error.message : "Unknown error",
-          );
-        }
-      } catch (err) {
-        console.error("Error delegating to worker:", err);
-      }
+  // When worker initializes, send the bot config to initialize the bot
+  cluster.on("message", (worker, message: WorkerMessage) => {
```

Comment:
Maybe it would be easier just to pass the token to the bot through env in the `fork` method?